qemu/tests/block_discard.py

```diff
@@ -114,11 +121,15 @@ def get_allocation_bitmap():
         logging.debug("bitmap before test: %s" % bitmap_before_trim)
         test.fail("bitmap should be continuous before fstrim")
 
-    format_disk_cmd = params["format_disk_cmd"]
-    format_disk_cmd = format_disk_cmd.replace("DISK", device_name)
-    error_context.context("format disk '%s' in guest" % device_name,
+    error_context.context("Create partition on '%s' in guest" % device_name,
                           logging.info)
-    session.cmd(format_disk_cmd)
+    session.cmd(params['create_partition_cmd'].replace("DISK", device_name))
+
+    if not utils_misc.wait_for(_check_disk_partitions_number, 30, step=3.0):
+        test.error('Failed to get 1 partitions on %s.' % device_name)
```

Comment:
` 1 partition`

Reply:
failed to get partition is fine, '1' is a little wired here.

Reply:
thanks, have updated.